### PR TITLE
Fix compilation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/pad_controller.rs [hunk 1]
```diff
@@ -7,14 +7,13 @@ use glib::IsA;
...
 use PadController;
+use std::ffi::CString;
 
 pub trait PadControllerExtManual: 'static {
-    #[cfg(any(feature = "v3_22", feature = "dox"))]
     fn set_action_entries(&self, entries: &[PadActionEntry]);
 }
 
 impl<O: IsA<PadController>> PadControllerExtManual for O {
-    #[cfg(any(feature = "v3_22", feature = "dox"))]
```
Comment (EPashkin) [inline src/pad_controller.rs]: IMHO this is needed:
`<method name="set_action_entries" c:identifier="gtk_pad_controller_set_action_entries" version="3.22">`

Comment (GuillaumeGomez) [reply src/pad_controller.rs]: The whole file is under this compilation condition, take a look at `src/lib.rs`. ;)

Comment (EPashkin) [reply src/pad_controller.rs]: Thanks, I missed that.

Comment (GuillaumeGomez) [reply src/pad_controller.rs]: You're welcome!

**Comment by GuillaumeGomez** [comment]
I forgot to tag people btw:

cc @EPashkin @sdroege

**Comment by EPashkin** [comment]
These errors IMHO legit,
```
error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<&std::string::String>` is not satisfied
  --> /home/travis/build/gtk-rs/gtk/src/pad_controller.rs:21:23
   |
21 |             .map(|e| (CString::new(&e.label), CString::new(&e.action_name)))
   |                       ^^^^^^^^^^^^ the trait `std::convert::From<&std::string::String>` is not implemented for `std::vec::Vec<u8>`
   |
   = help: the following implementations were found:
             <std::vec::Vec<T> as std::convert::From<&[T]>>
             <std::vec::Vec<T> as std::convert::From<&mut [T]>>
             <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
             <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
           and 5 others
   = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `&std::string::String`
   = note: required by `std::ffi::CString::new`
error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<&std::string::String>` is not satisfied
  --> /home/travis/build/gtk-rs/gtk/src/pad_controller.rs:21:47
   |
21 |             .map(|e| (CString::new(&e.label), CString::new(&e.action_name)))
   |                                               ^^^^^^^^^^^^ the trait `std::convert::From<&std::string::String>` is not implemented for `std::vec::Vec<u8>`
   |
   = help: the following implementations were found:
             <std::vec::Vec<T> as std::convert::From<&[T]>>
             <std::vec::Vec<T> as std::convert::From<&mut [T]>>
             <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
             <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
           and 5 others
   = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `&std::string::String`
   = note: required by `std::ffi::CString::new`
error[E0599]: no method named `as_ptr` found for type `&std::result::Result<std::ffi::CString, std::ffi::NulError>` in the current scope
  --> /home/travis/build/gtk-rs/gtk/src/pad_controller.rs:30:30
   |
30 |                 label: label.as_ptr() as *mut _,
   |                              ^^^^^^
error[E0599]: no method named `as_ptr` found for type `&std::result::Result<std::ffi::CString, std::ffi::NulError>` in the current scope
  --> /home/travis/build/gtk-rs/gtk/src/pad_controller.rs:31:42
   |
31 |                 action_name: action_name.as_ptr() as *mut _,
   |                                          ^^^^^^
```

**Comment by GuillaumeGomez** [comment]
They are indeed. It's surprising this whole file got merged...

**Comment by GuillaumeGomez** [comment]
Ok, fixed errors but I'd appreciate if something could double-check if the dangling pointers issue isn't back. :)

File: src/pad_controller.rs [hunk 2]
```diff
@@ -7,29 +7,28 @@ use glib::IsA;
...
     fn set_action_entries(&self, entries: &[PadActionEntry]) {
         let n_entries = entries.len() as i32;
         let entry_strings = entries
             .iter()
-            .map(|e| (CString::new(&e.label), CString::new(&e.action_name)))
-            .collect::<Vec<_>>();
+            .map(|e| (e.label.to_glib_none().0, e.action_name.to_glib_none().0))
+            .collect::<Vec<(*mut c_char, *mut c_char)>>();
```
Comment (EPashkin) [inline src/pad_controller.rs]: Now if  dangling pointers.
First vector must be owned or stashed `.to_glib_none()` without `.0`

Comment (EPashkin) [reply src/pad_controller.rs]: * Now we have  dangling pointers again

Comment (GuillaumeGomez) [reply src/pad_controller.rs]: Ok, kept the `Stashed` instead.

**Comment by EPashkin** [comment]
@GuillaumeGomez rustdoc-stripper 0.1.6 fails on gdk/src/enums.rs even in purge mode

**Comment by EPashkin** [comment]
and this seems somewhat related to lines count:
if I remove all consts after line 423 `pub const c: Key = gdk_sys::GDK_KEY_c as u32;` 
build script passed, same if I replace `c`-line with old next line `pub const d: Key = gdk_sys::GDK_KEY_d as u32;`

**Comment by EPashkin** [comment]
just adding `d`-line will cause stackoverflow

**Comment by GuillaumeGomez** [comment]
I'll fix the line count this evening.

**Comment by GuillaumeGomez** [comment]
Fixed.

**Comment by GuillaumeGomez** [comment]
cc @EPashkin @sdroege

File: src/pad_controller.rs [hunk 3]
```diff
@@ -9,27 +9,25 @@ use PadActionEntry;
 use PadController;
 
 pub trait PadControllerExtManual: 'static {
-    #[cfg(any(feature = "v3_22", feature = "dox"))]
     fn set_action_entries(&self, entries: &[PadActionEntry]);
```
Comment (sdroege) [inline src/pad_controller.rs]: Please also implement https://github.com/gtk-rs/gtk/pull/826#discussion_r293673105 and https://github.com/gtk-rs/gtk/pull/826#discussion_r293673403

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by sdroege** [comment]
:+1:

**Pull request merged**